### PR TITLE
Correct sync on metachain to resolve block body

### PR DESCRIPTION
**Pull request by sasurobert**

### CODE REVIEW

File: process/sync/metablock.go [hunk 1]
```diff
@@ -333,7 +366,63 @@ func (boot *MetaBootstrap) getMetaHeaderFromPool(headerHash []byte) (data.Header
...
+	if err != nil {
+		return nil, err
+	}
+
+	blockBody := block.Body(miniBlockSlice)
+
+	return blockBody, nil
+}
+
+func (boot *MetaBootstrap) requestMiniBlocksFromHeaderWithNonceIfMissing(_ uint32, nonce uint64) {
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/sync/metablock.go]: done.

File: cmd/node/config/economics.toml [hunk 2]
```diff
@@ -17,4 +17,4 @@
 
 [ValidatorSettings]
     StakeValue = "500000000000000000000000"
-    UnBoundPeriod = "100000"
+    UnBoundPeriod = "10"
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply cmd/node/config/economics.toml]: only for testing.

File: process/sync/metablock.go [hunk 3]
```diff
@@ -405,6 +405,8 @@ func (boot *MetaBootstrap) requestMiniBlocksFromHeaderWithNonceIfMissing(_ uint3
 		return
 	}
 
+	boot.requestedItemsHandler.Sweep()
+
 	hashes := make([][]byte, len(header.MiniBlockHeaders))
```
Comment (SebastianMarian) [inline process/sync/metablock.go]: You forgot to call Has from requestedItemsHandler in next for. You should replace lines between 410-413 with these:

	hashes := make([][]byte, 0)
	for i := 0; i < len(header.MiniBlockHeaders); i++ {
		if boot.requestedItemsHandler.Has(string(header.MiniBlockHeaders[i].Hash)) {
			continue
		}

		hashes = append(hashes, header.MiniBlockHeaders[i].Hash)
	}

Comment (sasurobert) [reply process/sync/metablock.go]: done.

File: integrationTests/sync/edgeCases_test.go [hunk 4]
```diff
@@ -20,6 +21,8 @@ func TestSyncMetaNodeIsSyncingReceivedHigherRoundBlockFromShard(t *testing.T) {
 		t.Skip("this is not a short test")
 	}
 
+	_ = logger.SetLogLevel("*:DEBUG")
```
Comment (iulianpascalau) [inline integrationTests/sync/edgeCases_test.go]: might remove this as running all tests in debug mode will output a plethora of messages.

File: process/peer/process.go [hunk 5]
```diff
@@ -122,13 +123,98 @@ func (p *validatorStatistics) IsNodeValid(node *sharding.InitialNode) bool {
...
+	zero := big.NewInt(0)
+	if peerChange.ValueChange.Cmp(zero) != 0 {
+		actualValue := zero.Add(account.Stake, peerChange.ValueChange)
+		err := account.SetStakeWithJournal(actualValue)
+		if err != nil {
+			return err
+		}
+	}
+
+	if peerChange.Action == block.PeerRegistrantion && peerChange.TimeStamp != account.Nonce {
```
Comment (iulianpascalau) [inline process/peer/process.go]: We use the timestamp as peer account's nonce? So counter intuitive.

Comment (sasurobert) [reply process/peer/process.go]: yeah, timeStamp is equal with nonce, maybe it should be round.

File: integrationTests/sync/edgeCases_test.go (hunk 4 above)
Comment (sasurobert) [reply integrationTests/sync/edgeCases_test.go]: done.

File: cmd/node/config/economics.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (sasurobert) [reply cmd/node/config/economics.toml]: done.

File: process/peer/process.go (hunk 5 above)
Comment (iulianpascalau) [reply process/peer/process.go]: It remains counter intuitive. Why didn't we have a specialized field called Timestamp or Round (just like Balance in the plain Account)?

Comment (SebastianMarian) [reply process/peer/process.go]: Can you put at least a TODO there to refactor this

**Pull request merged**